src/site/antora/modules/ROOT/pages/manual/webapp.adoc

```diff
@@ -15,311 +15,626 @@
+[source,jsp]
+----
+include::example$manual/webapp/taglib.jsp[tag=declaration]
+----
 
-When specifying these context parameters, you must specify them in the deployment descriptor (web.xml) even in a Servlet 3.0 or never application. If you add them to the ServletContext within a listener, Log4j will initialize before the context parameters are available and they will have no effect. Here are some sample uses of these context parameters.
+[WARNING]
+====
+The Log4j Taglib component is deprecated and is scheduled for removal in Log4j 3.
 
-==== Set the Logging Context Name to "myApplication"
+Currently, it only works with JavaServer Pages 2.3 or older and no version compatible with Jakarta Server Pages 3.0 is available.
+====
+
+The Log4j Taglib library defines a tag for most
+link:../javadoc/log4j-api/org/apache/logging/log4j/Logger.html[`Logger`]
+methods, including:
+
+* simple and parameterized log statements:
++
+[source,xml,indent=0]
+----
+include::example$manual/webapp/taglib.jsp[tag=simple]
+----
 
-[source,xml]
+* flow tracing statements:
++
+[source,xml,indent=0]
 ----
-<context-param>
-    <param-name>log4jContextName</param-name>
-    <param-value>myApplication</param-value>
-</context-param>
+include::example$manual/webapp/taglib.jsp[tag=entry-exit]
 ----
 
-==== Set the Configuration Path/File/URI to "/etc/myApp/myLogging.xml"
+* catching and throwing statements:
++
+[source,xml,indent=0]
+----
+include::example$manual/webapp/taglib.jsp[tag=catching]
+----
 
-[source,xml]
+* tags to test the current log level:
++
+[source,xml,indent=0]
 ----
-<context-param>
-    <param-name>log4jConfiguration</param-name>
-    <param-value>file:///etc/myApp/myLogging.xml</param-value>
-</context-param>
+include::example$manual/webapp/taglib.jsp[tag=if-enabled]
 ----
 
-[#use-jndi-context-selector]
-==== Use the JndiContextSelector
+* tags to set the name of the logger used:
++
+[source,xml,indent=0]
+----
+include::example$manual/webapp/taglib.jsp[tag=set-logger]
+----
 
-[source,xml]
+* a `dump` tag that prints the contents of a JSP scope:
++
+[source,xml,indent=0]
 ----
-<context-param>
-    <param-name>isLog4jContextSelectorNamed</param-name>
-    <param-value>true</param-value>
-</context-param>
-<context-param>
-    <param-name>log4jContextName</param-name>
-    <param-value>appWithJndiSelector</param-value>
-</context-param>
-<context-param>
-    <param-name>log4jConfiguration</param-name>
-    <param-value>file:///D:/conf/myLogging.xml</param-value>
-</context-param>
+include::example$manual/webapp/taglib.jsp[tag=dump]
 ----
 
-Note that in this case you must also set the "Log4jContextSelector" system property to "org.apache.logging.log4j.core.selector.JndiContextSelector".
+A complete JSP page example is available
+{antora-examples-url}/manual/webapp/taglib.jsp[on GitHub].
```

Comment:
While it is personal, I very much like disclosing the entire source in the block caption, for instance,

```
. Snippet from an example {antora-examples-url}/manual/webapp/taglib.jsp[`taglib.jsp`]
[source,xml,indent=0]
----
include::example$manual/webapp/taglib.jsp[tag=dump]
----
```

Reply:
Fixed in 2809eaae3db7089f9c56b9b1e5a218e0606a4f38